src/scheduler/resv_info.cpp

```diff
@@ -1710,7 +1711,7 @@ confirm_reservation(status *policy, int pbs_sd, resource_resv *unconf_resv, serv
 		snprintf(confirm_msg, LOG_BUF_SIZE, "%s:partition=%s", PBS_RESV_CONFIRM_SUCCESS,
 			 sc_attrs.partition ? sc_attrs.partition : DEFAULT_PARTITION);
 
-		pbsrc = pbs_confirmresv(pbs_sd, const_cast<char *>(nresv_parent->name.c_str()), short_xc,
+		pbsrc = pbs_confirmresv(get_svr_inst_fd(pbs_sd, nresv_parent->svr_inst_id), const_cast<char *>(nresv_parent->name.c_str()), short_xc,
```

Comment:
Please add a sched level wrapper function for pbs_confirmresv so that the caller doesn't have to worry about multi-server stuff. Refer to sched_ifl_wrappers.cpp

Reply:
I have actually looked at one of the functions send_attr_updates() inside sched_ifl_wrappers.cpp. This is still getting called as follows.
```
rc = send_attr_updates(get_svr_inst_fd(pbs_sd, resresv->svr_inst_id), resresv->name, pattr);
rc = send_attr_updates(get_svr_inst_fd(pbs_sd, job->svr_inst_id), job->name, job->job->attr_updates);

```
And also I see there are other calls like pbs_sigjob(), pbs_geterrmsg(), are also getting called in the similar fashion i.e. using get_svr_inst_fd().

Are you suggesting wrapping all of these functions like sig_job(), geterrmsg(), send_attr_updates() etc and internally call get_svr_inst_fd() ? Please clarify.

Reply:
Please look at the latest code, I recently refactored it to make things simpler for callers and also added a wrapper for pbs_sigjob. I missed out pbs_geterrmsg(), if you want you can take it up, or leave it since it's not really in the scope of your PR. But please do add a wrapper for pbs_confirmresv() since you are touching it.